dist/performance.js

```diff
@@ -84,6 +85,100 @@ function getWebVitalsJS() {
+    const regexForCheckGTmetrix = new RegExp(/.{1}userAgent.{1,100}(?:GTmetrix|gtmetrix.com).{2}/)
+    const regexForCheckPageSpeed = new RegExp(/.{1}userAgent.{1,100}(?:PageSpeed).{2}/)
+
+    //inline scripts check
+    Array.from(rawDoc.querySelectorAll('script:not([src])')).forEach(script => {
+        let scriptTagCode = script.innerHTML;
+        if (regexForCheckChromeLH.test(scriptTagCode)){
+            returnObj['detectUA-ChromeLH'] = true;
+        }
+        if(regexForCheckGTmetrix.test(scriptTagCode)){
+            returnObj['detectUA-GTmetrix'] = true;
+        }
+        if(regexForCheckPageSpeed.test(scriptTagCode)){
+            returnObj['detectUA-PageSpeed'] = true;
+        }
+    });
+
+    //check external scripts response bodies
+    if (testPropertyStringInResponseBodies(regexForCheckChromeLH)) {
+        returnObj['detectUA-ChromeLH'] = true;
+    }
+
+    if (testPropertyStringInResponseBodies(regexForCheckGTmetrix)) {
+        returnObj['detectUA-GTmetrix'] = true;
+    }
+
+    if (testPropertyStringInResponseBodies(regexForCheckPageSpeed)) {
+        returnObj['detectUA-PageSpeed'] = true;
+    }
+
+    //https://www.debugbear.com/blog/optimizing-web-vitals-without-improving-performance
+    //catch lcp animation / cls animation & overlay hack
+    const regexForCheckfadeInAnimation = new RegExp(/this.style.animation.{1,10}.fadein.{1,20}.forwards/)
+    Array.from(rawDoc.querySelectorAll('img')).forEach(el => {
+        let onloadVal = el.getAttribute('onload');
+        if (onloadVal !== null) {
+            if(regexForCheckfadeInAnimation.test(onloadVal)) {
+                returnObj['lcpAnimation'] = true;
+            }
+        }
+        let styleObj = el.style;
+        if (styleObj['pointer-events'] == 'none' &&
+                styleObj['position'] == 'absolute' &&
+                styleObj['width'] == '99vw' &&
+                styleObj['height'] == '99vh') {
+            returnObj['lcpOverlay'] = true;
+        }
+    });
+
+    //add logic for svg & body overlay
+    Array.from(rawDoc.querySelectorAll('svg')).forEach(svg => {
+        if (svg.clientHeight == 99999 &&
+                svg.clientWidth == 99999 &&
+                svg.clientLeft == 0 &&
+                svg.clientTop == 0) {
+            //additional check required
+            returnObj['lcpSvgOverlay'] = true;
+        }
+    });
+
+
+    //fid iframe hack
+    Array.from(document.getElementsByTagName('iframe')).forEach(iframeElement => {
+        let iframeTransparencyVal = iframeElement.getAttribute('transparency');
+        if (iframeTransparencyVal) {
+            styleObj = getComputedStyles(iframeElement, ['position','width','top','z-index','left','height']) ;
+            if(styleObj['position'] == 'absolute' &&
+                    styleObj['top'] == '0px' &&
+                    styleObj['left'] == '0px' &&
+                    styleObj['z-index'] == '999'){
```

Comment:
Similarly, this could be generalized to "is there a big iframe covering the page"

Reply:
`fidIframeOverlaySoft = iframe > 90% of viewport & is highest layer`